### ISSUE TITLE
High CPU usage on latest macOS while spawning update processes (Intel)

### ISSUE THREAD

**Issue by aronson** [powercord-org/powercord#595]
**Describe the bug**
When I launch powercord, an automatic update fires, or I press the "check update" button, the renderer process uses at least 100% of one CPU core and the application responsiveness drops to several seconds between UI re-renders. This gets worse as additional repositories are checked, and the issue compounds.

**To Reproduce**
- Launch on macOS Monterey 12.1 beta (untested on mainline), git version 2.30.1, latest powercord sources as of this release
- Have several plugins from external repositories (I can provide my setup if requested)
- Observe CPU usage and application responsiveness drop

**Expected behavior**
I looked at the source for the updater and the issue isn't obvious. I'd expect what it's doing to take nearly no time at all given I can run the commands myself in the directories manually in seconds. It must be some kind of threading contention.

**Desktop (please complete the following information):**
 - OS: macOS Monterey 12.1 beta
 - Powercord version: 1d2318bcd7c5e5a5230d8bf25813ed648cfd7b76

**Additional context**
 - [x] I checked if Discord and/or Powercord are up to date
 - [?] I made sure this is not related to an external plugin or a theme (it gets worse when you have more!)
 
I do TypeScript work in my professional life. I looked through the sources but couldn't figure out how to attach my IDE, Visual Studio Code, to the running process. If I could get a debugger on the sources I'm sure I could track down the issue for you. If you have any resources on how to do that I'd be grateful. I'll also reach out on Discord.

Here's the CPU usage as seen in my process manager.
<img width="1440" alt="Screen Shot 2021-12-05 at 3 02 17 AM" src="https://user-images.githubusercontent.com/1026348/144740574-47c8bb15-565f-4a30-a9bc-2a7d87d2de32.png">

**Comment by tigattack** [comment]
+1, I'm seeing the same thing. Other applications aren't noticably affected (at least in my case), but Discord slows to an absolute crawl (pretty much entirely frozen, view updates every ~5 seconds).

Big Sur 11.6, Discord Canary 0.0.279, PowerCord ab4d015.

**Comment by tigattack** [comment]
Still having the same issue on macOS Monterey 12.4, Discord Canary 0.0.284, Powercord cedd005.

**Comment by asportnoy** [comment]
I can also reproduce this issue. After running Powercord for a while, I've found that it starts running extremely slowly for some unknown reason and doesn't really stop until I restart. I thought it was due to some plugin having some crazy memory leak or something of the sort but I think that this issue is the cause.

As a temporary solution, I decreased the update interval so that it would occur less often. I also found that decreasing the concurrency limit to 1 makes a big difference when it does update. Also note that when you press the "check for updates" button, it runs all updates concurrently, which makes the app basically completely freeze.
